services/py-api/src/database/db_manager.py

```diff
@@ -0,0 +1,88 @@
+import os
+from typing import Annotated
+
+from fastapi import Depends
+from motor.motor_asyncio import AsyncIOMotorClient
+from pymongo import MongoClient
+from pymongo.errors import ConnectionFailure, OperationFailure, ConfigurationError
+from result import Err
+from structlog.stdlib import get_logger
+
+from src.utils import SingletonMeta
+
+LOG = get_logger()
+
+
+class DatabaseManager(metaclass=SingletonMeta):
+    """Creates a Thread-safe singleton Database manager. It provides a wrapper over the AsyncIOMotorClient and utils for
+    pinging the database. The singleton behaviour allows us to have one instance of the AsyncIOMotorClient which we
+    could safely use across our application through the interface of the DatabaseManager"""
+
+    DB_NAME = "TheHubDEV"
+
+    def __init__(self) -> None:
+        # The mongo client has a conn pool under the hood, and we set a min number of idle conn that the pool has
+        # to maintain, the default is 0. After maxIdleTimeMS the connection pool replaces the idle conn with a new one.
+        # I set a min number of conn according to the article below. The connection is also removed too fast, after
+        # around 30 sec, that's why I increased maxIdleTimeMS to 5 min.
+        # https://alexedwards.net/blog/configuring-sqldb
+        # https://www.mongodb.com/docs/languages/python/pymongo-driver/current/faq/#how-does-connection-pooling-work-in-pymongo-
+        self._client = AsyncIOMotorClient(host=os.environ["DATABASE_URL"], minPoolSize=25, maxIdleTimeMS=5 * 60 * 1000)
```

Comment:
Why would you have such a high `minPoolSize`? This is processing power which you're wasting on managing open connections especially when you know that you don't have such high traffic.

Reply:
I got it from this article: https://alexedwards.net/blog/configuring-sqldb
What number do you suggest?

Reply:
I skimmed and found no reference of min value there

Reply:
I suggest you let the client handle itself.

Reply:
We could set the minPoolSize to be 10% of the maxPoolSize: https://medium.com/@dhanushkasampath.mtr/what-are-the-default-values-for-hikari-connection-pool-if-we-do-not-override-in-application-properti-11932cdbe321

Reply:
It's up to you. As mentioned I would usually just have the defaults until proven ineffective.

Reply:
Made some changes in [c2f5f08](https://github.com/AUBGTheHUB/monolith/pull/716/commits/c2f5f08237741c90057394088566612c86c3f99c). Let's leave it like this, later I will make some performance testing to see the differences.